numpy-stubs/__init__.pyi

```diff
@@ -465,6 +468,14 @@ def fromiter(iterable: Iterator, dytpe: Type[_DType], count: int = ...) -> ndarr
 def fromstring(
     string: str, dtype: Type[_DType] = ..., count: int = ..., sep: str = ...
 ) -> ndarray: ...
+def histogramdd(
+    a: ndarray,
+    bins: Optional[Union[ndarray, Series, List, int]],
```

Comment:
I think we should maybe use [a protocol](https://mypy.readthedocs.io/en/stable/protocols.html) here instead of `Series`. A protocol describes which method a class should implement. For example, `Iterable` describes all classes that have an `__iter__` function. The protocol that we should use here instead of Series, should be [Sequence](https://docs.python.org/3/library/typing.html#typing.Sequence). We would need to check that `Series` really is a `Sequence`. A `Sequence` needs to have these functions: `__len__`, `__iter__`, `__contains__`, `__getitem__`, `__reversed__`, I think. Maybe only `__len__` and `__getitem__` actually. I would need to try it out.

I can understand though if you don't want to wait anymore on this getting merged. So maybe we'll do it in another PR.

Reply:
I agree with both your points -- I think we need to use `Sequence` more widely -- I just found that we really need that instead of `List` for the parameter types in `np.array`.  I also think we should push that out to a separate commit: this one's already too big.

Reply:
Added #81  so this can be tracked more easily.